tf_quant_finance/black_scholes/approximations/american_option.py

```diff
@@ -0,0 +1,306 @@
+# Lint as: python3
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Baron-Adesi Whaley approximation of the Black Scholes equation to calculate
+ the price of a batch of American options."""
+
+import numpy as np
+import tensorflow.compat.v2 as tf
+import tf_quant_finance.black_scholes.implied_vol_newton_root as implied_vol_newton
```

Comment:
also,
from google3.third_party.nomisma_quant_finance.black_scholes import  implied_vol_newton_root

this is because is not part of the library